### PR TITLE
use pathogen slug for create and list samples

### PR DESCRIPTION
**Pull request by phoenixAja**
### Summary:
- **What:** use the pathogen slug to return data according to pathogen slug, or create new samples associated with correct pathogen
- **Ticket:** [sc211484](https://app.shortcut.com/genepi/story/211484)
- **Env:** `<rdev link>`

### Checklist:
- [x] I merged latest `trunk`
- [x] I manually verified the change
- [x] I added labels to my PR
- [ ] I tested in multiple browsers
- [x] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/backend/aspen/api/views/samples.py [hunk 1]
```diff
@@ -73,7 +73,8 @@ async def list_samples(
         selectinload(Sample.uploaded_by),
         selectinload(Sample.collection_location),
         selectinload(Sample.accessions),
-    )
+        selectinload(Sample.pathogen),
+    ).filter(Sample.pathogen_id == pathogen.id)
```
Comment (jgadling) [inline src/backend/aspen/api/views/samples.py]: This PR looks good! Only question is whether this `list` endpoint behaves reasonably well when `sample.pathogen_id` is null

Comment (phoenixAja) [reply src/backend/aspen/api/views/samples.py]: oh that's right we will have some samples without pathogen in the interim.. I think the best case here would be to return an exception?

Comment (jgadling) [reply src/backend/aspen/api/views/samples.py]: ha, we can't return an exception since we want our endpoints to keep working properly until we update all samples!

Returning 'pathogen: null` in the response is fine for the interim, but I want to make sure that samples without pathogens are still listed.

Comment (phoenixAja) [reply src/backend/aspen/api/views/samples.py]: so actually if we do have a null pathogen_slug the `get_pathogen_slug` will return `SC2` as a default, and in `get_pathogen` we `raise ex.BadRequestException("Invalid pathogen slug")` if no pathogen was found, so i think we're set on this and won't get any wierd edge cases. I'll add another test to verify though

**Review by lvreynoso — APPROVED** [review]
lgtm

**Review by jgadling — CHANGES_REQUESTED** [review]
This isn't going to work properly for samples without pathogens.

**Comment by phoenixAja** [comment]
@jgadling if no pathogen is provided we default to SC2, and if no pathogen is found for a given slug we raise an exception, i added tests for those two cases

**Review by jgadling — APPROVED** [review]
